File: checkIO/Common_Words.js
"use strict";

function commonWords(first, second) {
	var intersect = [];
	for(var f of first.split(",")){
		for(var s of second.split(",")){
			if(f == s){intersect.push(f)};
		};
	};
	return String(intersect.sort());
};

var assert = require('assert');

if (!global.is_checking) {
    assert.equal(commonWords("hello,world", "hello,earth"), "hello", "Hello");
    assert.equal(commonWords("one,two,three", "four,five,six"), "", "Too different");
    assert.equal(commonWords("one,two,three", "four,five,one,two,six,three"), "one,three,two", "1 2 3");
    console.log("Coding complete? Click 'Check' to review your tests and earn cool rewards!");
}

File: checkIO/common-words.go
package main

import (
	"fmt"
	"strings"
)

func commonWords(first, second string) {
	frsl := strings.Split(first, ",")
	srsl := strings.Split(second, ",")

	for _, f := range frsl {
		for _, s := range srsl {
			fmt.Println(f, s)
		}
	}
}

func main() {
    commonWords("hello,world", "hello,earth")	//"hello"
	commonWords("one,two,three", "four,five,six")	//""
    commonWords("one,two,three", "four,five,one,two,six,three")	//"one,three,two"
}

/*
function commonWords(first, second) {
	var intersect = [];
	for(var f of first.split(",")){
		for(var s of second.split(",")){
			if(f == s){intersect.push(f)};
		};
	};
	return String(intersect.sort());
};
*/
